include/ignition/rendering/base/BaseGeometry.hh

```diff
@@ -87,12 +95,37 @@ namespace ignition
       if (material) this->SetMaterial(material, _unique);
     }
 
+    //////////////////////////////////////////////////
+    template <class T>
+    GeometryPtr BaseGeometry<T>::Clone() const
+    {
+      if (!this->Scene())
+      {
+        ignerr << "Cloning a geometry failed because the geometry to be "
+          << "cloned does not belong to a scene." << std::endl;
```

Comment:
Nit: prefer `\n` to std::endl;

Reply:
Interesting, I exclusively use `std::endl`. Is there an advantage to using `\n`?

Reply:
I'd also be curious to know about the difference. Is it because `\n` doesn't flush the buffer, which allows the `ignerr` stream to take care of that? I am pretty sure that we have a mixture of `\n` and `std::endl` being used with ign streams across ignition repositories